src/components/ProcessCreate/Questions/Question.tsx

```diff
@@ -74,4 +77,86 @@ const Question = ({ index, remove }: Props) => {
   )
 }
 
-export default Question
+interface CustomFieldError extends FieldError {
+  index: number
+}
+
+interface IQuestionPageProps {
+  title: string
+  description: string
+  isMultiQuestion?: boolean
+}
+
+const QuestionPage = ({ title, description, isMultiQuestion = false }: IQuestionPageProps) => {
+  const { t } = useTranslation()
+
+  const {
+    watch,
+    formState: { errors },
+  } = useFormContext()
+
+  const { fields, append, remove } = useFieldArray({
+    name: 'questions',
+  })
+
+  const [tabIndex, setTabIndex] = useState(0)
+
+  const questions = watch('questions')
+
+  useEffect(() => {
+    if (questions.length === 0) {
+      append({
+        title: '',
+        description: '',
+        options: [{ option: '' }, { option: '' }],
+      })
+    }
+
+    if (tabIndex === questions.length && tabIndex !== 0) setTabIndex(questions.length - 1)
+  }, [questions, append, tabIndex])
+
+  useEffect(() => {
+    const questionErrors = errors.questions as FieldErrors<CustomFieldError>[] | undefined
+
+    if (!questionErrors) return
+
+    const firstError = questionErrors.findIndex((curr) => typeof curr !== 'undefined')
+    setTabIndex(firstError)
+  }, [errors.questions])
```

Comment:
Friendly reminder to always comment effects :)

Reply:
Afair, I copied this effects from somewhere but i commented them anyway. Thanks for the reminder